extensions/wikia/WikiFactory/DeadWikis/maintenance.php

```diff
@@ -382,6 +402,85 @@ private function disableDiscussion( $cityId ) {
 		}
 	}
 
+	// Gets the most recent post from the specified site and returns whether it was made less than 60 days ago
+	private function isActiveSite($siteId)
+	{
+		$response = $this->getMostRecentPostForSite($siteId);
+
+		if ($response['postCount'] == 0) {
+			// no posts
+			return false;
+		}
+
+		// extract post creation date from response
+		$mostRecentPostCreationDate = $response['_embedded']->{'doc:posts'}[0]->{'creationDate'}->{'epochSecond'};
+
+		$sixtyDaysAgo = time() - 60*60*24*60;
+
+		return $mostRecentPostCreationDate > $sixtyDaysAgo;
+	}
+
+	private function getMostRecentPostForSite($siteId) {
+		$apiClient = $this->getSitesApi()->getApiClient();
+
+		$resourcePathTemplate = "/{siteId}/posts";
+		$httpBody = '';
+
+		// header params
+		$headerParams = ['Content-Type' => $apiClient->selectHeaderContentType(array('application/json'))];
+
+		$headerAccept = $apiClient->selectHeaderAccept(array('application/hal+json'));
+		if (!is_null($headerAccept)) {
+			$headerParams['Accept'] = $headerAccept;
+		}
+
+		// path params
+		$resourcePath = str_replace(
+				"{siteId}",
+				$apiClient->getSerializer()->toPathValue($siteId),
+				$resourcePathTemplate
+		);
+
+		// only need most recent post
+		$queryParams = [
+				'limit' => 1
+		];
+
+		// make the API Call
+		try {
+			list($rawResponse, $statusCode, $httpHeader) = $apiClient->callApi(
+					$resourcePath,
+					'GET',
+					$queryParams,
+					$httpBody,
+					$headerParams,
+					'object'
+			);
+			$response = $apiClient->getSerializer()->deserialize($rawResponse, 'object', $httpHeader);
+		} catch (\Swagger\Client\ApiException $e) {
+			throw $this->processApiException($e, $apiClient);
+		}
+
+		return $response;
+	}
+
+	private function processApiException($e, $apiClient) {
```

Comment:
Minor; could add PHPDoc at the function level or type hints in the method signature, e.g.:
```php
	/**
	 * @param Swagger\Client\ApiException $e
	 * @param Swagger\Client\ApiClient $apiClient
	 * @return mixed
	 */
```
or
```php
private function processApiException( Swagger\Client\ApiException $e, Swagger\Client\ApiClient $apiClient ) {
```
or
```php
/** @var Swagger\Client\ApiException $e */
```
This would allow easy navigation into the methods called in this function (currently the IDE doesn't know what type these variables are and so can't take you to the methods easily).